packages/amplify_api_plugin_interface/lib/src/GraphQL/GraphQLResponse.dart

```diff
@@ -13,12 +13,11 @@
  * permissions and limitations under the License.
  */
 
-import 'package:flutter/foundation.dart';
 import '../types.dart';
 
 class GraphQLResponse<T> {
   T data;
-  List<GraphQLResponseError> errors = [];
+  List<GraphQLResponseError>? errors;
 
-  GraphQLResponse({@required this.data, this.errors});
+  GraphQLResponse({required this.data, this.errors});
```

Comment:
Good point, it looks like all the points in which we call that constructor actually provide an errors = [] so it doesn't need to be nullable based on that utilization pattern. 

However the native side will not alway provide errors (since the operation could have completed smoothly).  Therefore it makes sense that errors could be nullable as they will not always exist.  Perhaps what we should do instead is to modify the method _deserializeGraphQLResponseErrors to be able to return null in case no errors were provided as parameters to it in the first place or to only call that method if errors from native side are not null. 

@Ashish-Nanda as this is your code, what are your thoughts on this?

Reply:
Good point, it looks like all the points in which we call that constructor actually provide an errors = [] so it doesn't need to be nullable based on that utilization pattern. 

However the native side will not alway provide errors (since the operation could have completed smoothly).  Therefore it makes sense that errors could be nullable as they will not always exist.  Perhaps what we should do instead is to modify the method _deserializeGraphQLResponseErrors to be able to return null in case no errors were provided as parameters to it in the first place or to only call that method if errors from native side are not null. 

@ashish-nanda as this is your code, what are your thoughts on this?